### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by BranBayou**
#### In this project I implemented the following requirements.

- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
     It will return the first user from the database.
     Don't worry about proper authentication at this time. You will add it in the upcoming projects.

- Create forms to perform the following functions:

     Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
    Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).

- Allow Users to add likes to Posts.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
   root "users#index"
-  resources :users, only: %i[show index]
-  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
-  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+  resources :users, only: [:show] do
+    resources :posts, only: [:index, :show]
+  end
+
+  resources :posts, only: [:new, :create] do
+    resources :comments, only: [:new, :create]
```
Comment (nahnahdev) [inline config/routes.rb:9]: - [ ] Great job with your implementation of the comments. However, per the project requirements, not only should be a pair of routes in the `config/routes.rb` file for creating comments but you should also ensure that there is a separate form in the `views/comments/new.html.erb` file that accepts input for creating comments. :+1:  So kindly implement this as well :+1:

File: config/routes.rb [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
+    resources :likes, only: [:create]
+  end
+
+
+
+  # resources :users, only: %i[show index]
+  # get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  # get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
```
Comment (nahnahdev) [inline config/routes.rb]: - [ ] Kindly avoid leaving commented pieces of code in your codebase. It makes your codebase unclean and is not a good practice :+1:

File: app/models/post.rb [hunk 3]
```diff
@@ -3,19 +3,16 @@ class Post < ApplicationRecord
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
 
-  validates :title, presence: true, length: { minimum: 2, maximum: 250 }
-  validates :comments_counter, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
-  validates :likes_counter, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :title, presence: true, length: { maximum: 250 }
+  # validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  # validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
Comment (nahnahdev) [inline app/models/post.rb]: - [ ] Kindly avoid leaving commented pieces of code in your codebase. It makes your codebase unclean and is not a good practice :+1:

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  # def create
+  #   user = User.find(params[:user_id])
+  #   post = Post.find(params[:post_id])
+  #   @comment = post.comments.new(author_id: current_user, **comment_param)
+  #   if @comment.save
+  #     redirect_to user_post_path(current_user, post)
+  #   else
+  #     redirect_to user_post_path(current_user), alert: 'Failed to add comment!'
+  #   end
+  # end
```
Comment (nahnahdev) [inline app/controllers/comments_controller.rb]: - [ ] Kindly avoid leaving commented pieces of code in your codebase. It makes your codebase unclean and is not a good practice :+1:

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @BranBayou  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- Linters well setup and passing :heavy_check_mark: 
- Professional README :heavy_check_mark: 
- ApplicationController contains `#current_user` method :heavy_check_mark: 
- Most required routes present :heavy_check_mark: 
- Most required forms present :heavy_check_mark: 
- User is able to add a like to posts :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @BranBayou , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Great job implementing the requested changes
:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file and PR

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**